### ISSUE TITLE
issue with "python-docs-samples/iap/example_startup_script.sh"

### ISSUE THREAD

**Issue by sminorff** [GoogleCloudPlatform/python-docs-samples#4260]
## In which file did you encounter the issue?

python-docs-samples/iap/example_startup_script.sh

## Describe the issue

<!-- Please be specific. Copying and pasting your invocation and the entire
I follow the IAP tutorial documentation[1] to test IAP. 
In the step 6 part, I should get the message like "Hi, user@example.com. I am my-managed-instance-group-29z6.". However, I have got a below error message. 
![error](https://user-images.githubusercontent.com/48852524/86903973-04797c80-c14b-11ea-944e-870e2b5b4e60.png)
-----------
Error: Server Error
The server encountered a temporary error and could not complete your request.
Please try again in 30 seconds.
-----------
[1] https://cloud.google.com/iap/docs/tutorial-gce#create_gce_template

**Comment by tmatsuo** [comment]
@sminorff 

Can you describe the issue?

**Comment by sminorff** [comment]
I follow the IAP tutorial documentation[1] to test IAP. 
In the step 6 part, I should get the message like "Hi, user@example.com. I am my-managed-instance-group-29z6.". However, I have got a below error message. 

I have tried to follow the documentation more than 10 times but it always has got a same message  
-----------
Error: Server Error
The server encountered a temporary error and could not complete your request.
Please try again in 30 seconds.
-----------
sorry, I am not good at python or some of program languages. However, I am sure the issue with the script. 
 
[1] https://cloud.google.com/iap/docs/tutorial-gce#create_gce_template

**Comment by tmatsuo** [comment]
@bshaffer Can you take a look?

**Comment by bshaffer** [comment]
@sminorff Can you go to the Logging UI in Cloud Console (http://console.cloud.google.com/logs) and filter by **VM Instance** > [your instance name] ? From here, you should be able to see an error being thrown.

**Comment by sminorff** [comment]
I am not able to find any error message stackdriver logging,firewall and LB

Please note that everything is fine and I can test IAP except the error message when successfully logged in with IAP.

It will only take 10 minutes to test it with the startup script.

**Comment by bshaffer** [comment]
I was able to launch an instance group with the startup script, and it worked as expected. If you're receiving a server error, a message should definitely show up in Cloud Logging.

One issue could be if the startup script was launched before you created `my-backend-service`, then the IAP validation will fail because the backend service ID didn't exist at the time the validation was added. Try deleting your instances and starting up new ones.
